packages/openapi-to-graphql/src/preprocessor.ts

```diff
@@ -699,11 +699,29 @@ export function createDataDef<TSource, TContext, TArgs>(
     // Found existing data definition and fetch it
     const existingDataDef = data.defs[index]
 
-    collapseLinksIntoDataDefinition({
-      additionalLinks: saneLinks,
-      existingDataDef,
-      data
-    })
+    /**
+     * Special handling for oneOf. Subdefinitions are always an array 
+     * (see createOneOfUnion)
+     */
+    if (
+      existingDataDef.targetGraphQLType === TargetGraphQLType.oneOfUnion && 
```

Comment:
I see what you were trying to do here but actually the OAS you provided doesn't make use of the code here. This is to collapse links in existing data definitions.